### ISSUE TITLE
When set Taks_Type as GPU, the training loss almost no improvement 

### ISSUE THREAD

**Issue by davidguo-7** [catboost/catboost#1614]
Problem: When set Task_Type as GPU, the training loss almost no improvement. But when I set Task_Type as CPU, the training loss decreases very fast. My dataset's shape is (214578, 104). I installed Catboost via pip. training loss function is MAE. Hope someone can help me. Thanks a lot!
catboost version: 0.24.4
Operating System: win10
CPU: AMD R9 4900
GPU: RTX 2060 Max Q

![image](https://user-images.githubusercontent.com/74689498/111727402-ccb00080-88a5-11eb-9ef1-00bd4fa816a1.png)
![image](https://user-images.githubusercontent.com/74689498/111727448-dfc2d080-88a5-11eb-98da-d7456dae191d.png)

**Comment by davidguo-7** [comment]
My Nvidia driver version is 461.92.

**Comment by ek-ak** [comment]
Hello! What learning parameters do you have? Can you try setting bootstrap_type='Bayesian'?

**Comment by davidguo-7** [comment]
> Hello! What learning parameters do you have? Can you try setting bootstrap_type='Bayesian'?
![image](https://user-images.githubusercontent.com/74689498/111855676-36411500-8961-11eb-9382-31c39fe8cff5.png)


Hi! I modified the training parameters as you suggested. But it still not works. I using Catboost for a regression task. Please check my parameters below:
model = cb.CatBoostRegressor(iterations = 5000,
                                        learning_rate = 0.03,
                                        depth= 9,
                                        l2_leaf_reg=5,
                                        loss_function='MAE',
                                        eval_metric= 'MAE',
                                        task_type = 'GPU',
                                        bootstrap_type = 'Bayesian'
                                        )
Besides, should I change any setting for  Nvidia driver or CUDA? I haven't done it yet.

**Comment by davidguo-7** [comment]
Hope anyone can help on the issue! thanks a lot!

**Comment by ek-ak** [comment]
Please try catboost  version 0.25  - Exact  leaf_estimation_method for GPU is available since 0.25. It should help you.

**Comment by davidguo-7** [comment]
Hi, any ideas on this error Option approx_on_full_history is unimplemented for task GPU?

**Comment by kizill** [comment]
@davidguo-7 Sorry, but I don't understand your reply. Did you try to setting `leaf_estimation_method='Exact'` as a training parameter for GPU?

**Comment by davidguo-7** [comment]
Hi Kizill, Yes. I tried it, but returned error message which mentioned that leaf_estimation_method='Exact' not support GPU training then my jupyter notebook crashed. I checked the official documentation, this point also be noted.
![image](https://user-images.githubusercontent.com/74689498/113668270-8ef40a00-96e4-11eb-8081-0734237f03a8.png)

**Comment by LyzhinIvan** [comment]
Hi, @davidguo-7! Are you sure that you use catboost v.0.25? Exact method is available on GPU since 0.25 version.

**Comment by LyzhinIvan** [comment]
Close because of no activity.
